gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/DistcpConverter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.converter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+
+import com.google.common.base.Function;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.Converter;
+import gobblin.converter.DataConversionException;
+import gobblin.converter.SchemaConversionException;
+import gobblin.converter.SingleRecordIterable;
+import gobblin.data.management.copy.CopyableFile;
+import gobblin.data.management.copy.FileAwareInputStream;
+import gobblin.util.PathUtils;
+
+
+/**
+ * Abstract class for distcp {@link Converter}. Simply transforms the {@link InputStream} in the
+ * {@link FileAwareInputStream}, and possibly modifies extensions of the output file.
+ */
+public abstract class DistcpConverter extends Converter<String, String, FileAwareInputStream, FileAwareInputStream> {
```

Comment:
Not required. Reduces code repetition (it's there because at some point in the development of the feature it was required, and I saw no reason to remove it).